checker/src/main/java/org/checkerframework/checker/units/UnitsRelationsTools.java

```diff
@@ -215,14 +215,12 @@ private static boolean hasNoPrefix(@Nullable final AnnotationValue annotationVal
         }
 
         if (hasNoPrefix(unitsAnnotation)) {
+            // Optimization, though the else case would also work.
             return unitsAnnotation;
         } else {
-            // In the Units Checker, the only annotation value is the prefix value.  Therefore,
-            // fromName (which creates an annotation with no values) is acceptable.
-            // TODO: refine sensitivity of removal for extension units, in case extension
-            // Annotations have more than just Prefix in its values.
-            return AnnotationBuilder.fromName(
-                    elements, unitsAnnotation.getAnnotationType().toString());
+            AnnotationBuilder builder = new AnnotationBuilder(elements, unitsAnnotation);
```

Comment:
I don't understand the motivation for this whole change. `fromName` returns a new annotation without any elements.
Why change this to a new constructor and then remove the element?

Reply:
I am trying to remove all uses of `fromName`, which is an error-prone interface.  If called with the name of an annotation that has elements/fields, then it silently produces an invalid annotation that can create problems (incorrect results, crashes) much later.